packages/relay/tests/redisInMemoryServer.ts

```diff
@@ -8,67 +28,62 @@ export class RedisInMemoryServer {
    *
    * @private
    */
-  private inMemoryRedisServer: RedisMemoryServer;
+  private readonly inMemoryRedisServer: RedisMemoryServer;
 
   /**
    * The logger used for logging all output from this class.
    * @private
    */
-  private logger: Logger;
-
-  /**
-   * The host of the in memory server.
-   * @private
-   */
-  private host: string;
-
-  /**
-   * The port of the in memory server.
-   * @private
-   */
-  private port: number;
+  private readonly logger: Logger;
 
   /**
    * The desired port to run on.
    * @private
    */
-  private portToRunOn: number;
+  private readonly portToRunOn: number;
 
   constructor(logger: Logger, portToRunOn: number) {
     this.logger = logger;
     this.portToRunOn = portToRunOn;
-  }
-
-  async start(): Promise<void> {
     this.inMemoryRedisServer = new RedisMemoryServer({
       instance: {
         port: this.portToRunOn,
       },
+      autoStart: false,
     });
+  }
 
-    this.port = await this.inMemoryRedisServer.getPort();
-    this.host = await this.inMemoryRedisServer.getHost();
+  async start(): Promise<void> {
+    this.logger.trace('Starting Redis in-memory server....');
+    const started = await this.inMemoryRedisServer.start();
+    if (started) {
+      const host = await this.getHost();
+      const port = await this.getPort();
+      this.logger.info(`Started Redis in-memory server on ${host}:${port} successfully.`);
+    } else {
+      this.logger.error("Couldn't start Redis in-memory server successfully.");
+    }
   }
```

Comment:
Could this benefit from a health check?  It might be handy if it fails in CI.
```
private async performHealthCheck(): Promise<void> {
  try {
    // Implement a simple health check, e.g., ping the server
    const client = await this.getClient();
    await client.ping();
    this.logger.debug('Redis in-memory server health check passed');
  } catch (error) {
    this.logger.warn(`Redis in-memory server health check failed: ${error.message}`);
  }
}
```

Reply:
There is no `pinging` in this case, because `ping()` is called on the side of the client to check if the server is running and the `RedisInMemoryServer` class is responsible for starting the in-memory server, so it's not a client. 

But I added something like a fallback "retry" mechanism, in case the redis in-memory server couldn't be started:
```typescript
  async start(): Promise<void> {
    this.logger.trace('Starting Redis in-memory server....');
    const started = await this.inMemoryRedisServer.start();
    if (started) {
      const host = await this.getHost();
      const port = await this.getPort();
      this.logger.info(`Started Redis in-memory server on ${host}:${port} successfully.`);
    } else {
      this.logger.error("Couldn't start Redis in-memory server successfully.");
      await this.performHealthCheck();
    }
  }

  private async performHealthCheck(): Promise<void> {
    try {
      // Ensure that the instance is running -> throws error if instance cannot be started
      const instanceData = await this.inMemoryRedisServer.ensureInstance();
      this.logger.debug(`Redis in-memory server health check passed, server is running on port ${instanceData.port}.`);
    } catch (error) {
      this.logger.warn(`Redis in-memory server health check failed: ${error}`);
    }
  }
```